_posts/2019-01-02-Deviceid-Reliability.md

```diff
@@ -0,0 +1,90 @@
+Every different device has a different y value to avoid overlaps between the points such that the visualization of the activity for each device is facilitated.
+
+
+* **Randomly generated deviceID**:<br/>
+Some app could provide a randomly generated deviceID.
+If this is the case, it is expected that the same deviceID is not observed in any other apps.
+We can think about two ways to generate the values:
+1) a new random deviceID is created for every acces;
+2) a particular transformation of the deviceID is used, thus the provided deviceID for the same real device is always the same but it will not be allined with the one provided by other apps.
+In both cases the fraction of devices shared with other apps should be zero.
+If we plot the histogram of this fraction we can see that there are no apps around zero.
+We can conclude that there are no apps in the data which generate a random deviceID and there is no need to include this selection in the global one.
+![unique_dev_random](../images/Deviceid-Reliability/unique_deviceid_random.png)
+
+
+Of course the same unreliable deviceID can be flagged by more than one pattern.
+However, there is not a single method among those which can be completely substituted by the others.
+Indeed, we include all of them in our deviceID reliability filter (except the randomly generated one since there are no devices which are randomly generated).
+
+**Put everything into production:**<br/>
+We put this quality filter selection into production creating a oozie pipeline which calls the hive queries that implements the described methods.
+The pipeline runs on a daily base such that as we acquire new data we filter out the broken one.
+To monitor the amount of the rejected devices and apps we store all the calculated information such that it is easy to trace back for which reason the data have been flagged as unreliable.  
+
+
+**Main benefits:**<br/>
+* Higher quality and reliability of your data mean higher correctness of business insights and better performances.
+In the end our customers wants to target some specific audience and the procedure to select that audience can only benefit from the removal of the broken data.
+You could think the data quality selection like the restoration of a piece of art.
+It is only when you remove the pollution effects that you can enjoy the original colors used by the artist.
+* The data quantity is reduced, so the computation time is speed up.
+Indeed, the quality filter is applied as soon as possible to optimize the overall general data flow. 
+* The quality selection is agnostic with respect to the model we apply. 
+Every process which use those data will benefit from the quality selection applied.
+You can think about a model which use the number of apps visited by the device as input feature.
+When you keep the fake devices which visit 1000 apps per day you should not be surprised if the model won't get the best out of that information.
```

Comment:
When you keep the fake devices which visit 1000 apps per day you should not be surprised if consuming processes won't get the best out of that information.